### PR TITLE
checks audience in everyone else rule

### PR DESCRIPTION
**Pull request by rashidsp**

### CODE REVIEW

File: lib/optimizely/decision_service.rb [hunk 1]
```diff
@@ -235,29 +235,19 @@ def get_variation_for_feature_rollout(feature_flag, user_id, attributes = nil)
...
-      return Decision.new(everyone_else_experiment, variation, DECISION_SOURCE_ROLLOUT) unless variation.nil?
-
+      # Check that user meets audience conditions for last rule
+      if Audience.user_in_experiment?(@config, everyone_else_experiment, attributes)
+        variation = @bucketer.bucket(everyone_else_experiment, bucketing_id, user_id)
+        return Decision.new(everyone_else_experiment, variation, DECISION_SOURCE_ROLLOUT) unless variation.nil?
+      end
       @config.logger.log(
         Logger::DEBUG,
         "User '#{user_id}' was excluded from the 'Everyone Else' rule for feature flag"
```
Comment (oakbani) [inline lib/optimizely/decision_service.rb]: Please change this message to: 
"User '#{user_id}' does not meet the conditions to be in rollout rule for audience '#{audience_name}'.

File: spec/decision_service_spec.rb [hunk 2]
```diff
@@ -561,41 +547,35 @@
...
-            expect(spy_logger).to have_received(:log).once
-                                                     .with(Logger::DEBUG, "User '#{user_id}' was excluded due to traffic allocation. Checking 'Everyone Else' rule now.")
           end
         end
       end
     end
 
     describe 'when the user is not bucketed into any targeting rules' do
-      it 'should try to bucket the user into the "Everyone Else" rule' do
+      it 'should bucket the user into the "Everyone Else" rule' do
```
Comment (oakbani) [inline spec/decision_service_spec.rb]: The previous was correct. I will still try to bucket. Won't bucket in all cases

File: spec/decision_service_spec.rb [hunk 3]
```diff
@@ -611,6 +591,47 @@
         expect(spy_logger).to have_received(:log).once
                                                  .with(Logger::DEBUG, "User '#{user_id}' does not meet the conditions to be in rollout rule for audience '#{audience_name}'.")
       end
+
+      it 'should not bucket the user into the "Everyone Else" rule' do
```
Comment (oakbani) [inline spec/decision_service_spec.rb]: Make it 
 it 'should not bucket the user into the "Everyone Else" rule when audience mismatch'

File: spec/decision_service_spec.rb [hunk 4]
```diff
@@ -611,6 +591,47 @@
...
+        everyone_else_experiment = rollout['experiments'][2]
+        variation = everyone_else_experiment['variations'][0]
+        allow(Optimizely::Audience).to receive(:user_in_experiment?).and_return(false)
+
+        allow(Optimizely::Audience).to receive(:user_in_experiment?)
+          .with(config, everyone_else_experiment, user_attributes)
+          .and_return(false)
+        allow(decision_service.bucketer).to receive(:bucket)
+          .with(everyone_else_experiment, user_id, user_id)
+          .and_return(variation)
```
Comment (oakbani) [inline spec/decision_service_spec.rb]: No need for these: 
```
+        allow(Optimizely::Audience).to receive(:user_in_experiment?)
+          .with(config, everyone_else_experiment, user_attributes)
+          .and_return(false)
+        allow(decision_service.bucketer).to receive(:bucket)
+          .with(everyone_else_experiment, user_id, user_id)
+          .and_return(variation)
```

**Comment by rashidsp** [comment]
@oakbani Please review changes.

**Pull request closed without merging**